onnxruntime/core/providers/dml/DmlExecutionProvider/src/Operators/DmlOperatorElementWise.cpp

```diff
@@ -560,13 +570,96 @@ class DmlOperatorElementwiseQLinear : public DmlOperator
+            opDesc.ScaleTensor = &inputDescs[OnnxInputIndex::scaleIndex];
+            opDesc.ZeroPointTensor = &inputDescs[OnnxInputIndex::zeroPointIndex];
+            opDesc.OutputTensor = &outputDescs[0];
 
-        SetDmlOperatorDesc({ApiTraits::OperatorDescTraits<TOperatorDesc>::Type, &opDesc}, kernelInfo);
+            SetDmlOperatorDesc({ApiTraits::OperatorDescTraits<TOperatorDesc>::Type, &opDesc}, kernelInfo);
+        }
+        // Create a zero point tensor, since it's a required input by DML
+        else
+        {
+            auto inputSizes = m_inputTensorDescs[0].GetSizes();
+            TensorDesc intermediateOutputTensorDesc = TensorDesc(
+                    GetMlDataTypeFromDmlDataType(inputDataType),
+                    inputSizes, 
+                    inputSizes,
+                    TensorAxis::DoNotCoerce,
+                    TensorAxis::W,
+                    TensorAxis::RightAligned,
+                    NchwDimensionCount, // minDimensionCount
+                    0 // guaranteedBaseOffsetAlignment
+                );
+            DML_TENSOR_DESC namedIntermediateOutputTensorDesc = intermediateOutputTensorDesc.GetDmlDesc();
+            
+            // Create a tensor full of zeros
+            DML_FILL_VALUE_CONSTANT_OPERATOR_DESC zerosDesc = {};
+            zerosDesc.ValueDataType = inputDataType;
+            zerosDesc.OutputTensor = &namedIntermediateOutputTensorDesc;
+            const DML_OPERATOR_DESC opDesc1 = { DML_OPERATOR_FILL_VALUE_CONSTANT, &zerosDesc };
+
+            TOperatorDesc qLinearDesc = {};
+            qLinearDesc.InputTensor = &inputDescs[OnnxInputIndex::inputIndex];
+            qLinearDesc.ScaleTensor = &inputDescs[OnnxInputIndex::scaleIndex];
+            qLinearDesc.ZeroPointTensor = zerosDesc.OutputTensor;
+            qLinearDesc.OutputTensor = &outputDescs[0];
+            const DML_OPERATOR_DESC opDesc2 = { ApiTraits::OperatorDescTraits<TOperatorDesc>::Type, &qLinearDesc};
+
+            // Construct the graph
+            std::vector<DML_INPUT_GRAPH_EDGE_DESC> inputEdges;
+            std::vector<DML_INTERMEDIATE_GRAPH_EDGE_DESC> intermediateEdges;
+            std::vector<DML_OUTPUT_GRAPH_EDGE_DESC> outputEdges;
+
+            MLOperatorGraphDesc operatorGraphDesc = {};
+            operatorGraphDesc.nodeCount = 2;
+            std::vector<const DML_OPERATOR_DESC*> opDescs{&opDesc1, &opDesc2};
+            operatorGraphDesc.nodesAsOpDesc = opDescs.data();
+
+            const uint32_t fillValueNodeIndex = 0;
+            const uint32_t dequantizeNodeIndex = 1;
+
+            // Input edges
+            DML_INPUT_GRAPH_EDGE_DESC inputToDequantizeEdge = {};
+            inputToDequantizeEdge.GraphInputIndex = OnnxInputIndex::inputIndex;
+            inputToDequantizeEdge.ToNodeIndex = dequantizeNodeIndex; 
+            inputToDequantizeEdge.ToNodeInputIndex = 0;
+            inputEdges.push_back(inputToDequantizeEdge);
```

Comment:
>inputEdges.push_back(inputToDequantizeEdge); [](http://example.com/codeflow?start=12&length=44)

Related to other comment, no need for heap allocated vector, since the size is a constant.

``
            DML_INPUT_GRAPH_EDGE_DESC inputEdges[2];

            // Input edges
            auto& inputToDequantizeEdge = inputEdges[0];
            inputToDequantizeEdge.GraphInputIndex = OnnxInputIndex::inputIndex;
            inputToDequantizeEdge.ToNodeIndex = dequantizeNodeIndex; 
            inputToDequantizeEdge.ToNodeInputIndex = 0;

            auto& scaleToDequantizeEdge = inputEdges[1];
            scaleToDequantizeEdge.GraphInputIndex = OnnxInputIndex::scaleIndex; // dmlWeightsIndex
            scaleToDequantizeEdge.ToNodeIndex = dequantizeNodeIndex; //dequantizeinputindex
            scaleToDequantizeEdge.ToNodeInputIndex = 1;

            operatorGraphDesc.inputEdgeCount = gsl::narrow_cast<uint32_t>(std::size(inputEdges));
            operatorGraphDesc.inputEdges = std::data(inputEdges);
```